index.js

```diff
@@ -27,113 +26,146 @@ let historyBus;
  *
  * @param  {String}     baseUrl          Base Path (to be ignored from URL.location)
  * @param  {String}     initialPath      Starting Path (should match one of your routes)
- * @param  {...Mixed}   routesAndReturns (String|Regex, Function, n+) Route + Function to call on match
+ * @param  {...*}       routesAndReturns (String|Regex, Function, n+) Route + Function to call on match
  * @return {Observable}                  EventStream that returns your matched route stream per route.
  */
 export default function baconRouter(baseUrl, initialPath, ...routesAndReturns) {
     // @TODO BaseUrl, Initial Path, And RoutesAndReturns should be objects.
     // Required next is '404' or missed routes perhaps. Currently the returned stream
     // is just 'nothing', means we won't hit anything.
 
+    // Because the routes and functions are 'paired', loop in increments of 2, first section is a route
+    // where the second section is the function to call and return.
+    // Generate [[(currentRoute) => [matches, value]]
+    const routesAndHandlers = routesAndReturns
+        .reduce((acc, routeOrReturn, i) => (
+            i % 2
```

Comment:
can you use [lodash/chunk](https://devdocs.io/lodash~4/index#chunk) instead?

Reply:
Done.